app/controllers/course_summaries_controller.rb

```diff
@@ -15,7 +15,15 @@ def index
 
   def populate
     if current_user.admin?
-      render json: get_table_json_data
+      table = JSON.parse(get_table_json_data)
+      data = (table[0]['assignment_marks'].map do |marks|
+
+          {
+            accessor: "assignment_marks.#{marks[0]}",
+            Header: Assignment.select("short_identifier").find(marks[0])['short_identifier']
+          }
+        end)
+      render json: {data: get_table_json_data,marks: data}
```

Comment:
Overall, you should pick better key names (both `data:` and `marks:`), and pick a better variable name than `data` too.